cdap-common/src/main/java/co/cask/cdap/common/HttpExceptionHandler.java

```diff
@@ -54,7 +54,9 @@ public void handle(Throwable t, HttpRequest request, HttpResponder responder) {
       responder.sendStatus(HttpResponseStatus.METHOD_NOT_ALLOWED);
     } else if (t instanceof UnauthorizedException) {
       logWithTrace(request, t);
-      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+      HttpResponseStatus status = ((UnauthorizedException) t).isAuthenticationError() ?
+        HttpResponseStatus.UNAUTHORIZED : HttpResponseStatus.FORBIDDEN;
```

Comment:
No, this is the right thing to do per HTTP standard - https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error.

401 if authentication error, 403 if authentication succeeds but authorization fails.